Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Location/AddEditDrawingsModal.tsx

```diff
@@ -0,0 +1,121 @@
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  11/06/2024 - Collins Self
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import { SystemCenter } from "@gpa-gemstone/application-typings";
+import { Input, Select } from "@gpa-gemstone/react-forms";
+import { Modal } from "@gpa-gemstone/react-interactive";
+import React from "react";
+
+interface IProps {
+    Record: SystemCenter.Types.LocationDrawing,
+    Setter: (record) => void,
+    HandleSave: () => void,
+    Show: boolean,
+    SetShow: (show: boolean) => void,
+}
+
+const AddEditDrawingsModal = (props: IProps) => {
+    const [category, setCategory] = React.useState<Array<SystemCenter.Types.ValueListItem>>([]);
+
+    function valid(field: keyof (SystemCenter.Types.LocationDrawing)): boolean {
+        if (field == 'Name')
+            return props.Record.Name != null && props.Record.Name.length > 0 && props.Record.Name.length <= 200;
+        else if (field == 'Link')
+            return props.Record.Link != null && props.Record.Link.length > 0;
+        else if (field == 'Number')
+            return props.Record.Number == null || props.Record.Number.length <= 50;
+        return true;
+    }
+
+    function getValueList(listName: string, setter: (value: Array<SystemCenter.Types.ValueListItem>) => void): JQuery.jqXHR<Array<SystemCenter.Types.ValueListItem>> {
+        let h = $.ajax({
+            type: "GET",
+            url: `${homePath}api/ValueList/Group/${listName}`,
+            contentType: "application/json; charset=utf-8",
+            dataType: `json`,
+            cache: false,
+            async: true
+        });
+        h.done((dCat: Array<SystemCenter.Types.ValueListItem>) => {
+            setter(dCat);
+
+        });
+        return h;
+    }
+
+    React.useEffect(() => {
+        let categoryHandle = getValueList("Category", setCategory);
+
+        return () => {
+            if (categoryHandle != null && categoryHandle.abort != null) categoryHandle.abort();
+        }
+    }, [])
+
+    return <>
+        <Modal
+            Show={props.Show}
+            Title={'Add/Edit Drawing'}
```

Comment:
see Confluence for Modal styling

Reply:
See [here](https://gridprotectionalliance.atlassian.net/wiki/spaces/TK/pages/202997764/UI+Standards+-+Buttons#Modals)

Reply:
ShowCancel has been set to false